src/main/java/com/minecolonies/core/entity/citizen/citizenhandlers/CitizenHappinessHandler.java

```diff
@@ -141,11 +141,12 @@ public double getHappiness(final IColony colony, final ICitizenData citizenData)
             double totalWeight = 0.0;
             for (final IHappinessModifier happinessModifier : happinessFactors.values())
             {
-                final double factor = happinessModifier.getFactor(citizenData);
-                if (factor == 1.0)
+                final double weight = happinessModifier.getWeight();
+                if (weight == 0.0)
```

Comment:
This was to filter out neutral factors (at 1.0) not to make it a mean. This is atm an average.

Reply:
Though a (weighted) factor of 1 should also be part of the average, otherwise the weights don't add up anymore in a good way

Reply:
(Sum(w*f)+Sum(w1*1))/(Sum(w)+Sum(w1)) is not equal to Sum(w*f)/Sum(w)

Reply:
> Though a (weighted) factor of 1 should also be part of the average, otherwise the weights don't add up anymore in a good way

We ignore them from the total weight as well, so it's fine. These are factors that "do currently not apply". They are removed intentionally. That could be something like "citizens are not unhappy with food because they are in a low level building" but this doesn't mean they're happy with it. They are just not unhappy with it.

Reply:
Are there enough factors initially to have a happiness at all? As currently, the very first citizens seem to spawn in with a happiness of 1. If everything starts out at default, then everything is stripped out until there is not much left to calculate happiness with, leaving it at a wrong value

Reply:
Is this also a problem in 1.20.1

Reply:
This is a recent problem, that was not a problem before.

Reply:
Yeah, I tested it in 1.20.1
Issue here https://github.com/ldtteam/minecolonies/issues/10320

Reply:
And yeah, it is a semi recent problem

Reply:
Ignoring them is I think the problem. A factor that doesn't influence the total is not a 1.-factor, but a factor with a weight of 0. A factor of 1. does influence the weighted average to tend to 1.
I'll be happy to try implementing this new behavior if you also think it's a good pattern.

Reply:
The way things were coded atm is that factors 1.0 only happens for neutral factors. E.g. mystical site non existent, school factors for non children, etc. Otherwise factors are always either > or < 1

If we reworked this, we'd add a special "don't apply" flag which is independent of factor and weight then.